packages/legend-query-builder/src/components/QueryBuilderPropertyExpressionEditor.tsx

```diff
@@ -205,6 +226,75 @@ export const QueryBuilderPropertyExpressionEditor = observer(
     const { propertyExpressionState } = props;
     const handleClose = (): void =>
       propertyExpressionState.setIsEditingDerivedProperty(false);
+    const queryBuilderState = propertyExpressionState.queryBuilderState;
```

Comment:
What were we doing before this change? were we resetting without the user confirmation?